tests/acceptance/api_test.go

```diff
@@ -1306,6 +1336,77 @@ func TestApi(t *testing.T) {
 		}, getResJSON["metadata"])
 	})
 
+	t.Run("Create multiple leases on one account", func(t *testing.T) {
```

Comment:
We specifically saw this bug when recreating a lease against the same account, multiple times. 

I did try recreating a lease 20 times while testing this out locally. It passed, but took some 15m to run. Generally I saw an error by attempt no 3. Still takes a couple minutes to run, but I think it's a worthwhile test to have in the arsenal.